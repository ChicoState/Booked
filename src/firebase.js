import * as firebase from 'firebase';
import 'firebase/auth';

var config = {
  apiKey: "AIzaSyAbrPremBQusjYoKqKfTcVHwxuw-Q8ikHw",
  authDomain: "booked-8586e.firebaseapp.com",
  databaseURL: "https://booked-8586e.firebaseio.com",
  projectId: "booked-8586e",
  storageBucket: "booked-8586e.appspot.com",
  messagingSenderId: "24720035765",
  appId: "1:24720035765:web:3d57fd5c840b8100c353fc",
  measurementId: "G-20QKEY7V1D"
};

firebase.initializeApp(config);

// var provider = new firebase.auth.GoogleAuthProvider();
// //var user = firebase.auth().currentUser;
// var name, email;
//
//
// firebase.auth().useDeviceLanguage();
// firebase.auth().onAuthStateChanged(function(user){
//
// 	if(user != null){
// 		name = user.displayName;
// 		email = user.email;
// 		alert(name + ': ' + email);
// 	}
// 	else{
// 		alert('Working on Redirect for initial log in');
// 		firebase.auth.signinWithRedirect(provider);
// 		firebase.auth().getRedirectResult().then(function(result) {
// 			if (result.credential) {
// 				//This gives you a Google Access Token.
// 				//var token = result.credential.accessToken;
// 			}
// 			user = result.user;
// 		}).catch(function(error) {
// 				// Handle Errors here.
// 				var errorCode = error.code;
// 				var errorMessage = error.message;
// 				// The email of the user's account used.
// 				var errorEmail = error.email;
// 				// The firebase.auth.AuthCredential type that was used.
// 				var credential = error.credential;
//
// 				//Output line for error variables to silence unused var warnings
// 				alert(errorCode + errorMessage + errorEmail + credential);
// 		});
// 	}
// });
