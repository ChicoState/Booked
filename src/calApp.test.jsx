import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
//import configureMockStore from 'redux-mock-store';
//import thunk from 'redux-thunk';
//import { Provider } from 'react-redux';
import CalApp from './calApp';
import FirebaseApp from './FirebaseApp'
import firebase from 'firebase';

// const mockStore = configureMockStore([thunk]);
// const store = mockStore();
// const FirebaseAuthProps = {};


jest.mock('firebase', () => {
  const data = { name: "text" };
  const snapshot = { val: () => data };
  const ref = jest.fn().mockReturnThis();
  const on = jest.fn(() => Promise.resolve(snapshot));
  const database = jest.fn();
  database.Database = jest.fn().mockReturnValue({ref,on});

  const auth = jest.fn();

  const mAuth = { signInWithRedirect: jest.fn() };
  auth.GoogleAuthProvider = jest.fn();
  auth.Auth = jest.fn(() => mAuth);
  return { auth, database };
});

describe('<CalApp />', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
const wrapper = mount(<CalApp />);
const wrapper2 = mount(<CalApp />);
const wrapper3 = mount(<CalApp />);
//wrapper.update();
if(wrapper.find('outty') > 0){
  wrapper.find('outty').simulate("click");
}

it('renders without crashing', () => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  const div = document.createElement('div');
  ReactDOM.render(<CalApp />, div);
});


it('render with dataload', done => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  const div = document.createElement('div');
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  const itemsRef = FirebaseApp.database().ref('text');
  itemsRef.on('value', (snapshot) => {
    let items = snapshot.val();
     try {
       expect(items).toStrictEqual(expect.anything());
       done();
     } catch (error) {
       done(error);
     }
  });
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  wrapper3.setState({isSignedIn:true});
  ReactDOM.render(wrapper3, div);
  wrapper3.setState({cuid:'tGVt2mPFHXTEh9xl40SQEirzYaU2'});
  wrapper3.update();
//  wrapper3.update();
  wrapper3.setState({isSignedIn:false});
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  wrapper3.setState({isSignedIn:true});
  ReactDOM.render(<CalApp/>, div);
});

it('async comparison', () => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  const div = document.createElement('div');
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  wrapper3.setState({isSignedIn:true});
  ReactDOM.render(wrapper3, div);
  wrapper3.setState({cuid:'tGVt2mPFHXTEh9xl40SQEirzYaU2'});
  wrapper3.update();
//  wrapper3.update();
  wrapper3.setState({isSignedIn:false});
  wrapper3.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
  wrapper3.setState({isSignedIn:true});
  expect(ReactDOM.render(<CalApp/>, div)).toStrictEqual(expect.anything());
});

// it('clickcolors', () => {
//   const wrapper4 = shallow(<CalApp />);
//   const instance = wrapper4.instance();
//   wrapper4.setState({cuid:'7JmSHENe6vebq1H5JOP3n0VObsD3'});
// //  const event = wrapper4.find("#.fc-event-main");
// //  expect(event.simulate("click")).toStrictEqual(expect().anything;
//   expect(wrapper4.find("fc-event-main")).toStrictEqual(expect().anything;
// });


it('render when signing in and out', done => {
  firebase.auth.mockImplementation(() => new firebase.auth.Auth());
  wrapper2.setState({cuid:'tGVt2mPFHXTEh9xl40SQEirzYaU2'});
  const itemsRef = FirebaseApp.database().ref('text');
  itemsRef.on('value', (snapshot) => {
    let items = snapshot.val();
     try {
       expect(items).toStrictEqual(expect.anything());
       done();
     } catch (error) {
       done(error);
     }
  });
  wrapper2.setState({cuid:'tGVt2mPFHXTEh9xl40SQEirzYaU2'});
  const div = document.createElement('div');
  ReactDOM.render(wrapper2, div);
  wrapper2.update();
  wrapper2.setState({cuid:'tGVt2mPFHXTEh9xl40SQEirzYaU2'});
  wrapper2.update();
  wrapper2.setState({isSignedIn:false});
  wrapper2.setState({isSignedIn:true});
  ReactDOM.render(<CalApp />, div);
});


// test('compare output', done => {
//   firebase.auth.mockImplementation(() => new firebase.auth.Auth());
//   firebase.auth.currentUser = true;
//   const itemsRef = FirebaseApp.database().ref('text');
//   itemsRef.on('value', (snapshot) => {
//     let items = snapshot.val();
//      try {
//        expect(items).toStrictEqual(expect().anything;
//        done();
//      } catch (error) {
//        done(error);
//      }
//   });
//
//   const component = renderer.create(<CalApp />,);
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });


});


// import React from 'react';
// import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
// import CalApp from './calApp';
//
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<CalApp />, div);
// });
//
// test('compare output', () => {
//   const component = renderer.create(
//     <CalApp />,
//   );
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
