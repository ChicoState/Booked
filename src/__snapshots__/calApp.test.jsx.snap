// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalApp /> compare output 1`] = `<div />`;

exports[`compare output 1`] = `<div />`;
